python/social_bot/envs/grocery_ground.py

```diff
@@ -0,0 +1,232 @@
+                data=obs_data_space, sentence=DiscreteSequence(128, 24))
+            self._action_space = gym.spaces.Dict(
+                control=control_space, sentence=DiscreteSequence(128, 24))
+        else:
+            self.observation_space = obs_data_space
+            self.action_space = control_space
+
+    def reset(self):
+        """
+        Reset the environment
+        Args: None
+        Returns: Observaion of the first step
+        """
+        self._world.reset()
+        self._world.step(5)
+        self._collision_cnt = 0
+        self._cum_reward = 0.0
+        self._steps_in_this_episode = 0
+        self._goal = self._world.get_agent(self._goal_name)
+        self._teacher.reset(self._agent, self._world)
+        teacher_action = self._teacher.teach("")
+        if self._with_language:
+            obs_data = self._get_observation()
+            obs = OrderedDict(data=obs_data, sentence=teacher_action)
+        else:
+            obs = self._get_observation()
+        return obs
+
+    def _get_observation(self):
+        if self._use_image_obs:
+            obs_data = np.array(
+                self._agent.get_camera_observation(
+                    "default::pr2::pr2::head_tilt_link::head_mount_prosilica_link_sensor"
+                ),
+                copy=False)
+            obs_data = PIL.Image.fromarray(rgb2gray(obs_data)).resize(
+                self._resized_image_size, PIL.Image.ANTIALIAS)
+
+            obs_data = np.reshape(
+                np.array(obs_data),
+                [self._resized_image_size[0], self._resized_image_size[1], 1])
+        else:
+            objects_poses = []
+            objects_poses.append(self._goal.get_pose())
+            objects_poses.append(self._agent.get_pose())
+            obs_data = np.array(objects_poses).reshape(-1)
+        return obs_data
+
+    def step(self, action):
+        """
+        Args:
+            action (dict|int): If with_language, action is a dictionary with key "control" and "sentence".
+                    action['control'] is a vector whose dimention is
+                    len(_joint_names). action['sentence'] is a string.
+                    If not with_language, it is an int for the action id.
+        Returns:
+            If with_language, it is a dictionary with key 'data' and 'sentence'
+            If not with_language, it is a numpy.array or image for observation
+        """
+        if self._with_language:
+            sentence = action.get('sentence', None)
+            controls = action['control']
+        else:
+            sentence = ''
+            controls = action
+        controls = dict(zip(self._pr2_joints, controls))
+        teacher_action = self._teacher.teach(sentence)
+        self._agent.take_action(controls)
+        self._world.step(5)
+        if self._with_language:
+            obs_data = self._get_observation()
+            obs = OrderedDict(data=obs_data, sentence=teacher_action.sentence)
```

Comment:
How is sentence used by RL algorithm? Right now, it a string which does not match the specification given at line 137